logstash-core/lib/logstash/api/modules/settings.rb

```diff
@@ -0,0 +1,36 @@
+# encoding: utf-8
+module LogStash
+  module Api
+    module Modules
+      class Settings < ::LogStash::Api::Modules::Base
+        def handle_logging(logging_keys, settings)
+          logging_keys.each do |key|
+            level = settings.delete(key)
```

Comment:
I am uneasy about modifying the `settings` object here (with delete). Can you tell me why you use delete, here?